libbeat/cmd/instance/beat.go

```diff
@@ -307,9 +307,15 @@ func NewBeatReceiver(settings Settings, receiverConfig map[string]interface{}, u
 
 	if settings.DisableConfigResolver {
 		config.OverwriteConfigOpts(obfuscateConfigOpts())
-	} else {
+	} else if store != nil {
 		// TODO: Allow the options to be more flexible for dynamic changes
 		config.OverwriteConfigOpts(configOpts(store))
+	} else {
+		config.OverwriteConfigOpts([]ucfg.Option{
```

Comment:
This branch confuses me -- it looks like it's just resetting the config opts back to their (current) default value, so can't we just skip the else block? (And similar comment in the same code below). If there's some reason an override is explicitly needed here, I'd rather have it be a helper function in the config package, e.g. `config.ResetConfigOpts()`, instead of manually duplicating the values here.

Reply:
Possible alternative here, after a bit more thought: if the else case is important, and you take the other suggestion to rename `configOpts` to `configOptsWithKeystore`, then you could instead make the options for this case a sibling helper function called `configOptsWithoutKeystore`. That way the reasoning for each branch is clearly spelled out.